Sources/NIOFileSystem/Internal/SystemFileHandle.swift

```diff
@@ -1046,6 +1046,70 @@ extension SystemFileHandle: WritableFileHandleProtocol {
             try sendableView._resize(to: size).get()
         }
     }
+
+    public func setTimes(
+        lastAccessTime: FileInfo.Timespec?,
+        lastDataModificationTime: FileInfo.Timespec?
+    ) async throws {
+        try await self.threadPool.runIfActive { [sendableView] in
+            try sendableView._withUnsafeDescriptor { descriptor in
+                if lastAccessTime == nil, lastDataModificationTime == nil {
+                    // If the timespec array is nil, as per the `futimens` docs,
+                    // both the last accessed and last modification times
+                    // will be set to now.
+                    futimens(descriptor.rawValue, nil)
+                } else {
+                    let lastAccessTimespec: timespec
+                    if let lastAccessTime = lastAccessTime {
+                        lastAccessTimespec = timespec(
+                            tv_sec: lastAccessTime.seconds,
+                            tv_nsec: lastAccessTime.nanoseconds
+                        )
+                    } else {
+                        // Don't modify the last access time.
+                        // Note: tv_sec will be ignored.
+                        lastAccessTimespec = timespec(
+                            tv_sec: 0,
+                            tv_nsec: Int(UTIME_OMIT)
+                        )
+                    }
+
+                    let lastDataModificationTimespec: timespec
+                    if let lastDataModificationTime = lastDataModificationTime {
```

Comment:
FWIW:  rather than all the if-elses we can switch over both values to create a `timespec`:

```swift
let ts: timespec

switch (lastAccessTime, lastModificationTime) {
case (.some(let lastAccess), .some(let lastModification)):
  ts = timspec(...)
case (.some(let lastAccess), .none):
  // ...
case (.none, .some(let lastModification)):
  // ...
case (.none, .none):
  // ...
}
```